sdk/core/azure-core/test/ut/transport_adapter_curl.cpp

```diff
@@ -40,11 +40,13 @@ namespace Azure { namespace Core { namespace Test {
     Azure::Core::Http::CurlConnectionPool::ClearIndex();
 
     auto threadRoutine = [&]() {
+      using namespace std::chrono_literals;
       auto request = Azure::Core::Http::Request(Azure::Core::Http::HttpMethod::Get, host);
       auto response = m_pipeline->Send(Azure::Core::GetApplicationContext(), request);
       checkResponseCode(response->GetStatusCode());
       auto expectedResponseBodySize = std::stoull(response->GetHeaders().at("content-length"));
       CheckBodyFromBuffer(*response, expectedResponseBodySize);
+      std::this_thread::sleep_for(1s);
```

Comment:
This test change seems unrelated to this PR. Why was it failing and why do we need to sleep here?

Reply:
It could have been added in a separate PR, yes. But it was really a super tiny thing!  hahaha

The test started to give some occasionally fails. The test needs to spawn 3 connections by sending 3 requests (one after the other). However, if the first request is completed too fast, then the 3rd request could re-use that connection instead of spawning a new one.  That's why 1 sec sleep will prevent that.
It is just a workaround to make the request to take at least 1 sec to be completed.  
Since this test is not really about if the request is succeeded or not, but to test that a new connection is spawn every time, I didn't invest on making the request adding a payload of something to make it take more time to complete.

Reply:
Gotcha. That explanation makes sense. Thanks!

nit: whenever we are modifying this file again later, please add this as a comment in the code which indicates why we need the sleep, in that PR.